main.js

```diff
@@ -294,6 +321,25 @@ export class IdDecoder {
         bitLength: this.fixedBitLength
       });
     }
+    if(this.multihash) {
+      // <varint hash fn code>: identity function
+      const [hashFnCode] = decoded.slice(0, 1);
+      if(hashFnCode !== MULTIHASH_IDENTITY_FUNCTION_CODE) {
+        throw new Error('Invalid multihash function code.');
+      }
+      // <varint digest size in bytes>: 32
+      const [digestSize] = decoded.slice(1, 2);
+
+      if(digestSize !== this.expectedSize) {
+        throw new Error('Invalid digest size.');
+      }
+      const keySeedBytes = decoded.slice(2, decoded.length);
```

Comment:
Could probably avoid the `slice` copy by using something like:
```suggestion
      const keySeedBytes = new Uint8Array(decoded.buffer, 2, decoded.length - 2);
```

Reply:
You can use `.subarray()` instead of `slice()` to be guaranteed to not make a copy:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray

```suggestion
      const keySeedBytes = decoded.subarray(2);
```

Reply:
Addressed 30d05b7